### DISCUSSION TITLE
请问2.0.0版本大概什么时候发布？

### DISCUSSION THREAD

**Discussion by monkeyWie** [apolloconfig/apollo#4257] [Q&A] [answered]
就差apollo我们就能迁移到java17了，期待ing!

**Comment by nobodyiam** [comment, marked as answer]
I think we could consider releasing the 2.0.0 version after #3866 is merged.

**Comment by dragontree101** [comment]
支持一下， 我们也就差apollo就可以整体升级到jdk17了

**Reply by nisiyong** [reply]
@nobodyiam I think maybe we could separate apollo-java-client from the main repo in the future.

**Reply by dragontree101** [reply]
> @nobodyiam I think maybe we could separate apollo-java-client from the main repo in the future.

这个建议不错， 这样客户端和服务端分离之后， 有一些项目的升级会比较简单一点，
另外，我们还是希望2.0.0能够快速发布， 我看所有的问题都解决完了，谢谢各位大佬。

**Reply by nobodyiam** [reply]
> @nobodyiam I think maybe we could separate apollo-java-client from the main repo in the future.

This sounds a good idea.

**Reply by Anilople** [reply]
@nobodyiam @nisiyong And split apollo-client to raw part and spring part?

* raw part: user can use it without spring framework
* spring part: user can integration apollo-client if run with spring framework. i.e some module like apollo-client-spring-boot-starter

**Reply by nobodyiam** [reply]
@Anilople 

Currently, users could use `apollo-client` with or without spring framework, as the spring related dependencies are optional. Do you mean to add a new project `spring-boot-starter-apollo` like what #1657 proposed?  I think [apollo-client-config-data](https://www.apolloconfig.com/#/zh/usage/java-sdk-user-guide?id=_3214-spring-boot-config-data-loader-spring-boot-24-apollo-client-190-%e6%8e%a8%e8%8d%90) may have already done part of that?

**Reply by Anilople** [reply]
> Do you mean to add a new project spring-boot-starter-apollo like what https://github.com/apolloconfig/apollo/issues/1657 proposed?

>> yes, and it will regular the code by denpendency, then we doesn't need to write code carefully to forbid it will use spring's dependency. see https://github.com/apolloconfig/apollo/pull/3937 

> I think [apollo-client-config-data](https://www.apolloconfig.com/#/zh/usage/java-sdk-user-guide?id=_3214-spring-boot-config-data-loader-spring-boot-24-apollo-client-190-%e6%8e%a8%e8%8d%90) may have already done part of that?

>> yes

---

The compatibility is also important, like maven dependency's artifactId

**Reply by nobodyiam** [reply]
> yes, and it will regular the code by denpendency, then we doesn't need to write code carefully to forbid it will use spring's dependency. see https://github.com/apolloconfig/apollo/pull/3937

This is a good point.

**Comment by ivan-lysytsia** [comment]
Hi @nobodyiam! 
Is there any updates on release date for 2.0 version?
It blocks my project from upgrade to Java 17.

**Reply by nisiyong** [reply]
We might release 2.0.0-RC1 recently, but you should know community has no responsibility to release for any company.

**Comment by nobodyiam** [comment]
@monkeyWie @dragontree101 @ivan-lysytsia 

Apollo 2.0.0-RC1 was just released. You may check out the [release note](https://github.com/apolloconfig/apollo/releases/tag/v2.0.0-RC1) for more information.

**Reply by monkeyWie** [reply]
This is awesome!

**Reply by dragontree101** [reply]
🎉🎉🎉🎉
